.github/workflows/auto-e2e-stress-test.yml

```diff
@@ -0,0 +1,92 @@
+name: Auto E2E Stress Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - "**/*.cy.spec.js"
+      - "**/*.cy.spec.ts"
+jobs:
+  detect-and-run-tests:
+    name: Detect and Run Stress Tests
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      pull-requests: write
+      # Needed to trigger workflows
+      actions: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Prepare frontend environment
+        uses: ./.github/actions/prepare-frontend
+
+      - name: Find and run affected tests
+        run: |
+          echo "Running cypress-test-finder against base branch: $BASE_BRANCH"
+          echo "Using head ref: $HEAD_REF"
+
+          # Run with --execute to automatically trigger workflows
+          node ./.github/scripts/cypress-test-finder.js --base="origin/$BASE_BRANCH" --ref="$HEAD_REF" --execute
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BASE_BRANCH: ${{ github.event.pull_request.base.ref }}
+          HEAD_REF: ${{ github.head_ref || github.ref_name }}
+
+      - name: Add or Update PR comment
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
```

Comment:
there's enough here, I would probably put it in a file and just pass it the github object